CONTRIBUTING.rst

```diff
@@ -41,7 +41,7 @@ Before you submit a pull request, make sure to complete the following steps:
 
    .. code-block:: console
 
-      $ pip install black pytype autoflake
+      $ pip install -r requirements-dev.txt
```

Comment:
what about:

```shell
poetry install --only dev
```

Reply:
We have an issue that Poetry requires Python version compliance with all dev dependencies and will not allow specification of a different Python version.

There's an in progress PR to add support for this, but it won't be production-ready for a while: https://github.com/python-poetry/poetry/pull/5740

Several dev dependencies have not been updated for Python 3.11, most notably pytype which requires python `>=3.7.1<3.11`.

Until that PR lands, an alternative solution is to extract development CLI dependencies into another file. Thoughts?